File: app/src/main/java/com/example/karatdatamobile/main/ReportsFragment.java
package com.example.karatdatamobile.main;

import android.annotation.SuppressLint;
import android.app.AlertDialog;
import android.app.Dialog;
import android.content.DialogInterface;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.DefaultItemAnimator;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.example.karatdatamobile.R;

import java.util.ArrayList;
import java.util.List;

public class ReportsFragment extends Fragment {
    TextView textReports;
    final String[] dialogArray = {"Открыть", "Редактировать", "Поделиться"};
    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View view = (ViewGroup) inflater.inflate(R.layout.fragment_reports, container, false);

        textReports = view.findViewById(R.id.report);
        textReports.setOnLongClickListener(new View.OnLongClickListener(){
            @SuppressLint("ResourceType")
            @Override
            public boolean onLongClick(View v) {
                AlertDialog.Builder builder = new AlertDialog.Builder(getActivity());
                builder.setTitle("Выберите действие");
                builder.setItems(dialogArray, new DialogInterface.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialogInterface, int i) {

                    }
                });
                AlertDialog alertDialog = builder.create();
                alertDialog.show();
                return true;
            }
        });
        return view;
    }
}

File: app/src/main/java/com/example/karatdatamobile/reports/ReportsFragment.kt
package com.example.karatdatamobile.reports

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.LinearLayoutManager
import com.example.karatdatamobile.databinding.FragmentReportsBinding
import moxy.MvpAppCompatFragment
import moxy.presenter.InjectPresenter
import moxy.presenter.ProvidePresenter
import toothpick.Toothpick

class ReportsFragment : MvpAppCompatFragment(), ReportsView {
    private var _binding: FragmentReportsBinding? = null
    private val binding get() = _binding!!

    @InjectPresenter
    lateinit var presenter: ReportsPresenter

    @ProvidePresenter
    fun providePresenter(): ReportsPresenter =
        Toothpick.openRootScope().getInstance(ReportsPresenter::class.java)


    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
         _binding = FragmentReportsBinding.inflate(inflater, container, false)
        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        init()
    }

    private fun init(){
        binding.recyclerView.layoutManager = LinearLayoutManager(context)
        binding.recyclerView.adapter = presenter.getAdapter()

    }
}